packages/manager/apps/dedicated/client/app/ip/components/list/list.controller.js

```diff
@@ -702,5 +706,78 @@ export default class IpListController {
     };
 
     refreshTable();
+
+    $scope.refreshIpList = (removedIps, createdIps, parentIpBlock) => {
+      const parentIpBlockRow =
+        parentIpBlock &&
+        $scope.ipsList.find((item) => item.ipBlock === parentIpBlock.ipBlock);
+
+      removedIps.forEach((ip) => {
+        const row = $scope.ipsList.find((item) => item.ipBlock === ip.ipBlock);
+        row.tobeRemoved = true;
+      });
+
+      const targetList = parentIpBlock ? parentIpBlockRow.ips : $scope.ipsList;
+      createdIps.forEach((ip) => {
+        targetList.push({
+          ...ip,
+          ip: ip.ipBlock,
+          block: ip.ipBlock,
+          isTemporaryEntry: true,
+        });
+      });
+
+      $scope.refreshQueue(removedIps, createdIps, parentIpBlockRow);
+    };
+
+    $scope.refreshQueue = (removedIps, createdIps, parentIpBlockRow) => {
```

Comment:
Can you please add some comments to understand the logic ? for both refreshQueue and refreshIpList